### ISSUE TITLE
[Bug]: Qwen3-next enable mtp with 0.12.0rc1 ops-error occurred.

### ISSUE THREAD

**Issue by Guozhong23** [vllm-project/vllm-ascend#5043]
### Your current environment

当前环境：
cann-8.3rc2

Package                           Version             Editable project location
--------------------------------- ------------------- ---------------------------
absl-py                           2.3.1
aiofiles                          25.1.0
aiohappyeyeballs                  2.6.1
aiohttp                           3.13.2
aiosignal                         1.4.0
annotated-doc                     0.0.4
annotated-types                   0.7.0
anthropic                         0.71.0
anyio                             4.12.0
asc_opc_tool                      0.1.0
astor                             0.8.1
attrs                             25.4.0
audioread                         3.1.0
auto_tune                         0.1.0
blake3                            1.0.8
blinker                           1.9.0
cachetools                        6.2.3
cbor2                             5.7.1
certifi                           2025.11.12
cffi                              2.0.0
charset-normalizer                3.4.4
click                             8.3.1
cloudpickle                       3.1.2
cmake                             4.2.0
compressed-tensors                0.12.2
Cython                            3.2.1
dataflow                          0.0.1
decorator                         5.2.1
depyf                             0.20.0
dill                              0.4.0
diskcache                         5.6.3
distro                            1.9.0
dnspython                         2.8.0
docstring_parser                  0.17.0
einops                            0.8.1
email-validator                   2.3.0
fastapi                           0.124.4
fastapi-cli                       0.0.16
fastapi-cloud-cli                 0.6.0
fastar                            0.8.0
filelock                          3.20.0
Flask                             3.1.2
frozenlist                        1.8.0
fsspec                            2025.12.0
gguf                              0.17.1
h11                               0.16.0
h2                                4.3.0
hccl                              0.1.0
hccl_parser                       0.1
hf-xet                            1.2.0
hpack                             4.1.0
httpcore                          1.0.9
httptools                         0.7.1
httpx                             0.28.1
huggingface-hub                   0.36.0
Hypercorn                         0.18.0
hyperframe                        6.1.0
idna                              3.11
interegular                       0.3.3
itsdangerous                      2.2.0
Jinja2                            3.1.6
jiter                             0.12.0
jmespath                          1.0.1
joblib                            1.5.2
jsonschema                        4.25.1
jsonschema-specifications         2025.9.1
lark                              1.2.2
lazy_loader                       0.4
librosa                           0.11.0
llguidance                        1.3.0
llm_datadist                      0.0.1
llm_datadist_v1                   0.0.1
llvmlite                          0.46.0
lm-format-enforcer                0.11.3
loguru                            0.7.3
markdown-it-py                    4.0.0
MarkupSafe                        3.0.3
mdurl                             0.1.2
mistral_common                    1.8.6
model-hosting-container-standards 0.1.11
modelscope                        1.33.0
mpmath                            1.3.0
msgpack                           1.1.2
msgspec                           0.20.0
msobjdump                         0.1.0
multidict                         6.7.0
networkx                          3.6.1
ninja                             1.13.0
numba                             0.63.1
numpy                             1.26.4
op_compile_tool                   0.1.0
op_gen                            0.1
op_test_frame                     0.1
opc_tool                          0.1.0
openai                            2.11.0
openai-harmony                    0.0.8
opencv-python-headless            4.11.0.86
outlines_core                     0.2.11
packaging                         25.0
pandas                            2.3.3
pandas-stubs                      2.3.3.251201
partial-json-parser               0.2.1.1.post7
pathlib2                          2.3.7.post1
pillow                            12.0.0
pip                               25.3
platformdirs                      4.5.1
pooch                             1.8.2
priority                          2.0.0
prometheus_client                 0.23.1
prometheus-fastapi-instrumentator 7.1.0
propcache                         0.4.1
protobuf                          6.33.2
psutil                            7.1.3
py-cpuinfo                        9.0.0
pybase64                          1.4.3
pybind11                          3.0.1
pycountry                         24.6.1
pycparser                         2.23
pydantic                          2.12.5
pydantic_core                     2.41.5
pydantic-extra-types              2.10.6
Pygments                          2.19.2
python-dateutil                   2.9.0.post0
python-dotenv                     1.2.1
python-json-logger                4.0.0
python-multipart                  0.0.20
pytz                              2025.2
PyYAML                            6.0.3
pyzmq                             27.1.0
Quart                             0.20.0
ray                               2.48.0
referencing                       0.37.0
regex                             2025.11.3
requests                          2.32.5
rich                              14.2.0
rich-toolkit                      0.17.0
rignore                           0.7.6
rpds-py                           0.30.0
safetensors                       0.7.0
schedule_search                   0.0.1
scikit-learn                      1.8.0
scipy                             1.15.3
sentencepiece                     0.2.1
sentry-sdk                        2.47.0
setproctitle                      1.3.7
setuptools                        65.5.0
setuptools-scm                    9.2.2
shellingham                       1.5.4
show_kernel_debug_data            0.1.0
six                               1.17.0
sniffio                           1.3.1
soundfile                         0.13.1
soxr                              1.0.0
starlette                         0.50.0
supervisor                        4.3.0
sympy                             1.14.0
te                                0.4.0
threadpoolctl                     3.6.0
tiktoken                          0.12.0
tokenizers                        0.22.1
torch                             2.8.0+cpu
torch_npu                         2.8.0
torchvision                       0.23.0+cpu
tqdm                              4.67.1
transformers                      4.57.1
triton-ascend                     3.2.0.dev2025110716
typer                             0.20.0
types-pytz                        2025.2.0.20251108
typing_extensions                 4.15.0
typing-inspection                 0.4.2
tzdata                            2025.2
urllib3                           2.5.0
uvicorn                           0.38.0
uvloop                            0.22.1
vllm                              0.12.0+empty        /vllm-workspace/vllm
vllm_ascend                       0.12.0rc1           /vllm-workspace/vllm-ascend
watchfiles                        1.1.1
websockets                        15.0.1
Werkzeug                          3.1.4
wheel                             0.45.1
wsproto                           1.3.2
xgrammar                          0.1.27
yarl                              1.22.0

### 🐛 Describe the bug

启动脚本：
`export HCCL_EXEC_TIMEOUT=204
export HCCL_CONNECT_TIMEOUT=120
export HCCL_IF_IP=80.48.5.68
export VLLM_ASCEND_ENABLE_NZ=0
export PYTORCH_NPU_ALLOC_CONF="expandable_segments:True"
export TRITON_ALL_BLOCKS_PARALLEL=1
export HCCL_OP_RETRY_ENABLE="L0:0, L1:0, L2:0"

# export MASTER_PORT=29605
# export MASTER_ADDR=127.0.0.1

export TASK_QUEUE_ENABLE=1
export CPU_AFFINITY_CONF=0
export HCCL_OP_EXPANSION_MODE="AIV"

export VLLM_WORKER_MULTIPROC_METHOD=spawn
#export ASCEND_RT_VISIBLE_DEVICES="8,9,10,11,12,13,14,15"
export ASCEND_RT_VISIBLE_DEVICES="0,1,2,3,4,5,6,7"

vllm serve ../Qwen3-Next-80B-A3B-Instruct/  \
 --port 8005   --host 0.0.0.0  \
 --served-model-name Qwen3-Next-80B-A3B-Instruct  \
 --tensor-parallel-size 8  \
 --max-model-len 37000  \
 --gpu-memory-utilization 0.65   \
 --max_cudagraph_capture_size 128 \
 --enforce-eager \
 --speculative-config '{"num_speculative_tokens": 2, "method":"qwen3_next_mtp"}'
 
#--compilation-config='{"cudagraph_mode":"FULL_DECODE_ONLY"}' \
# --enforce-eager
#--compilation-config '{"cudagraph_capture_sizes":[1,8,45,128],"cudagraph_mode":"FULL_DECODE_ONLY"}' \
`

测试场景：4k-32k-45并发，当kvcache利用率接近1的时候会报错。

[W1215 11:30:10.136265096 NPUStream.cpp:545] Warning: NPU warning, error code is 507035[Error]:
[Error]: The vector core execution is abnormal.
        Rectify the fault based on the error information in the ascend log.
EZ9999: Inner Error!
EZ9999[PID: 55986] 2025-12-15-11:30:10.420.394 (EZ9999):  [AIC_INFO] after execute:args print end[FUNC:GetError][FILE:stream.cc][LINE:1191]
        TraceBack (most recent call last):
       Aicore kernel execute failed, device_id=0, stream_id=2, report_stream_id=2, task_id=31025, flip_num=554, fault kernel_name=fused_recurrent_g          ated_delta_rule_fwd_kernel_46, fault kernel info ext=fused_recurrent_gated_delta_rule_fwd_kernel, program id=480, hash=15313142709528210228.[FUNC:G          etError][FILE:stream.cc][LINE:1191]
       rtDeviceSynchronizeWithTimeout execute failed, reason=[vector core exception][FUNC:FuncErrorReason][FILE:error_message_manage.cc][LINE:53]
       wait for compute device to finish failed, runtime result = 507035.[FUNC:ReportCallError][FILE:log_inner.cpp][LINE:162]
 (function npuSynchronizeUsedDevices)
terminate called after throwing an instance of 'c10::Error'
[W1215 11:30:10.138080186 NPUStream.cpp:545] Warning: NPU warning, error code is 507035[Error]:
[Error]: The vector core execution is abnormal.
        Rectify the fault based on the error information in the ascend log.
EZ9999: Inner Error!
EZ9999[PID: 55987] 2025-12-15-11:30:10.421.233 (EZ9999):  [AIC_INFO] after execute:args print end[FUNC:GetError][FILE:stream.cc][LINE:1191]
        TraceBack (most recent call last):
       Aicore kernel execute failed, device_id=1, stream_id=2, report_stream_id=2, task_id=31024, flip_num=554, fault kernel_name=fused_recurrent_g          ated_delta_rule_fwd_kernel_46, fault kernel info ext=fused_recurrent_gated_delta_rule_fwd_kernel, program id=480, hash=1852789123503244417.[FUNC:Ge          tError][FILE:stream.cc][LINE:1191]
       rtDeviceSynchronizeWithTimeout execute failed, reason=[vector core exception][FUNC:FuncErrorReason][FILE:error_message_manage.cc][LINE:53]
       wait for compute device to finish failed, runtime result = 507035.[FUNC:ReportCallError][FILE:log_inner.cpp][LINE:162]
 (function npuSynchronizeUsedDevices)
  what():  query:torch_npu/csrc/core/npu/NPUEvent.cpp:59 NPU function error: acl::AclQueryEventRecordedStatus(event_, &currStatus), error code is 5          07035
[ERROR] 2025-12-15-11:30:10 (PID:55986, Device:0, RankID:-1) ERR00100 PTA call acl api failed
[Error]: The vector core execution is abnormal.
        Rectify the fault based on the error information in the ascend log.
EH9999: Inner Error!
        rtEventQueryStatus execute failed, reason=[vector core exception][FUNC:FuncErrorReason][FILE:error_message_manage.cc][LINE:53]
EH9999[PID: 55986] 2025-12-15-11:30:10.425.676 (EH9999):  [Query][Status]query event recorded status failed, runtime result = 507035[FUNC:ReportCal          lError][FILE:log_inner.cpp][LINE:162]
        TraceBack (most recent call last):

Exception raised from query at torch_npu/csrc/core/npu/NPUEvent.cpp:59 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x80 (0x7f6e324          d34d0 in /usr/local/python3.11.13/lib/python3.11/site-packages/torch/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocat          or<char> > const&) + 0x65 (0x7f6e3246ff69 in /usr/local/python3.11.13/lib/python3.11/site-packages/torch/lib/libc10.so)
frame #2: c10_npu::NPUEvent::query() const + 0x39c (0x7f6e1eb1edfc in /usr/local/python3.11.13/lib/python3.11/site-packages/torch_npu/lib/libtorch_          npu.so)
frame #3: <unknown function> + 0xbd741c (0x7f6e1eab241c in /usr/local/python3.11.13/lib/python3.11/site-packages/torch_npu/lib/libtorch_npu.so)
frame #4: at_npu::native::CachingHostAllocator_emptyCache() + 0x51 (0x7f6e1eab2741 in /usr/local/python3.11.13/lib/python3.11/site-packages/torch_n          pu/lib/libtorch_npu.so)
frame #5: THPModule_npu_shutdown(_object*, _object*) + 0xa6 (0x7f6e2e1883c6 in /usr/local/python3.11.13/lib/python3.11/site-packages/torch_npu/_C.c          python-311-x86_64-linux-gnu.so)
<omitting python frames>
frame #17: <unknown function> + 0x29d90 (0x7f6e485b5d90 in /lib/x86_64-linux-gnu/libc.so.6)
frame #18: __libc_start_main + 0x80 (0x7f6e485b5e40 in /lib/x86_64-linux-gnu/libc.so.6)
frame #19: _start + 0x25 (0x55d7749b4095 in VLLM::Worker_TP0)

terminate called after throwing an instance of 'c10::Error'
  what():  query:torch_npu/csrc/core/npu/NPUEvent.cpp:59 NPU function error: acl::AclQueryEventRecordedStatus(event_, &currStatus), error code is 5          07035
[ERROR] 2025-12-15-11:30:10 (PID:55987, Device:1, RankID:-1) ERR00100 PTA call acl api failed
[Error]: The vector core execution is abnormal.
        Rectify the fault based on the error information in the ascend log.
EH9999: Inner Error!
        rtEventQueryStatus execute failed, reason=[vector core exception][FUNC:FuncErrorReason][FILE:error_message_manage.cc][LINE:53]
EH9999[PID: 55987] 2025-12-15-11:30:10.427.462 (EH9999):  [Query][Status]query event recorded status failed, runtime result = 507035[FUNC:ReportCal          lError][FILE:log_inner.cpp][LINE:162]
        TraceBack (most recent call last):

Exception raised from query at torch_npu/csrc/core/npu/NPUEvent.cpp:59 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) + 0x80 (0x7f754ba          d34d0 in /usr/local/python3.11.13/lib/python3.11/site-packages/torch/lib/libc10.so)
frame #1: c10::detail::torchCheckFail(char const*, char const*, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocat          or<char> > const&) + 0x65 (0x7f754ba6ff69 in /usr/local/python3.11.13/lib/python3.11/site-packages/torch/lib/libc10.so)
frame #2: c10_npu::NPUEvent::query() const + 0x39c (0x7f7538059dfc in /usr/local/python3.11.13/lib/python3.11/site-packages/torch_npu/lib/libtorch_          npu.so)
frame #3: <unknown function> + 0xbd741c (0x7f7537fed41c in /usr/local/python3.11.13/lib/python3.11/site-packages/torch_npu/lib/libtorch_npu.so)
frame #4: at_npu::native::CachingHostAllocator_emptyCache() + 0x51 (0x7f7537fed741 in /usr/local/python3.11.13/lib/python3.11/site-packages/torch_n          pu/lib/libtorch_npu.so)
frame #5: THPModule_npu_shutdown(_object*, _object*) + 0xa6 (0x7f75477c33c6 in /usr/local/python3.11.13/lib/python3.11/site-packages/torch_npu/_C.c          python-311-x86_64-linux-gnu.so)
<omitting python frames>
frame #17: <unknown function> + 0x29d90 (0x7f7561bf4d90 in /lib/x86_64-linux-gnu/libc.so.6)
frame #18: __libc_start_main + 0x80 (0x7f7561bf4e40 in /lib/x86_64-linux-gnu/libc.so.6)
frame #19: _start + 0x25 (0x56122b975095 in VLLM::Worker_TP1)

**Comment by Yikun** [comment]
@drslark Would you mind taking a look?

BTW, Please refer to https://docs.vllm.ai/projects/ascend/en/latest/tutorials/Qwen3-Next.html this should be stable way to run qwen next now.

**Comment by SunnyLee151064** [comment]
We have identified that this operator has an address out-of-bounds issue and are in the process of resolving it.

**Comment by wxsIcey** [comment]
I've tested on v0.13.0rc1, and this issue is not reproduced, could you plz try the  v0.13.0rc1 vllm-ascend?

**Comment by LoganJane** [comment]
@Guozhong23 This issue has been resolved in the latest version. Please use the latest release or rc version (https://github.com/vllm-project/vllm-ascend/releases). If the issue persists, feel free to submit a new issue.
